### PR TITLE
sdl_viewer: Terrain shaders

### PR DESCRIPTION
**Pull request by nnmm**
This program is going to run always for the same mesh: a regular grid with spacing 1. What's varying is the textures and uniforms.

### CODE REVIEW

**Comment by nnmm** [comment]
ptal @krzesi-mir

File: sdl_viewer/shaders/terrain.vs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// Essentially, the terrain regular grid with resolution terrain_res_m and origin
+// terrain_origin_m, and the mesh is rendering a rectangle within that grid.
+uniform dvec2 terrain_pos;
+// These two are equal and set by the GlMovingWindowTexture.
+// They hold the coordinates (in pixels) of the min corner
+// of the texture within the sampler.
+uniform ivec2 height_texture_offset;
+uniform ivec2 color_texture_offset;
+// The full projective transform.
+uniform dmat4 world_to_gl;
```
Comment (krzesi-mir) [inline sdl_viewer/shaders/terrain.vs]: nit: I don't know about this repo, but in `avsoftware` we have the `x_from_y` convention for transforms: `gl_from_world`?

File: sdl_viewer/shaders/terrain.vs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#version 410 core
+// The vertex positions are in grid coordinates, i.e. between 0 and grid_size (inclusive).
+layout(location = 0) in ivec3 aPos;
+
+// These are constant throughout the program.
+uniform double grid_size;
```
Comment (krzesi-mir) [inline sdl_viewer/shaders/terrain.vs]: is `grid_size` used anywhere? Also, should it have `int` type?

File: sdl_viewer/shaders/terrain.vs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  vec4 color;
+  uint quads;
+} vs_out;
+
+void main() {
+  // Each vertex position (aPos) maps to a pixel in the texture, but
+  // shifted (by height_texture_offset) with wraparound.
+  ivec2 texCoord = height_texture_offset + aPos.xy;
+  ivec2 texSize = textureSize(height, 0);
+  ivec2 texCoordModSize = texCoord - (texCoord / texSize) * texSize;
```
Comment (krzesi-mir) [inline sdl_viewer/shaders/terrain.vs]: can we use https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/mod.xhtml here?

Comment (nnmm) [reply sdl_viewer/shaders/terrain.vs]: That's for floats, unfortunately

File: sdl_viewer/shaders/terrain.vs (hunk 2 above)
Comment (nnmm) [reply sdl_viewer/shaders/terrain.vs]: No, good point!

File: sdl_viewer/shaders/terrain.vs (hunk 1 above)
Comment (nnmm) [reply sdl_viewer/shaders/terrain.vs]: I think that's nicer too, but wanted to keep it consistent with the rest of point_cloud_viewer, where it uses `y_to_x`.

File: sdl_viewer/shaders/terrain.vs (hunk 3 above)
Comment (krzesi-mir) [reply sdl_viewer/shaders/terrain.vs]: i see

**Pull request merged**